source/fundamentals/builders/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1198 @@
+       | `switchMapOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#switchMapOn(java.util.function.Function)>`__
+       | `switchNumberOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlNumber.html#switchNumberOn(java.util.function.Function)>`__
+       | `switchStringOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#switchStringOn(java.util.function.Function)>`__
+
+     - :manual:`$switch </reference/operator/aggregation/switch/>`
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+Suppose you have a collection of customers with their membership information.
+Originally, customers were either members or not. Over time, membership levels
+were introduced and used the same field. The information stored in this field
+can be one of a few different types, and you want to create a standardized value
+indicating their membership level.
+
+The ``switchOn()`` method runs through possible clauses without assuming the
+type of the ``member`` field. Once the ``switchOn()`` method determines the
+data type of the field, it calculates a string value corresponding to the
+membership level. If the data type is a string, it returns the membership
+level value. If the data type is a boolean, it returns either a ``Gold`` or
+``Guest`` membership level. If the data type is an array, it returns the
+most recent membership level, which is the last item in the array. If the
+``member`` field is an unknown type, the ``switchOn()`` method provides a
+default value of ``Guest``.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   asList(project(fields(
+            computed("membershipLevel",
+               current().getField("member").switchOn(field -> field
+                     .isString(s -> s)
+                     .isBoolean(b -> b.cond(of("Gold"), of("Guest")))
+                     .<MqlString>isArray(a -> a.last())
+                     .defaults(d -> of("Guest"))))
+   )));
+
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [
+     {
+       $project: {
+         result: {
+           $switch: {
+             branches: [
+               { case:
+                 { $eq: [ { $type: "$member" }, "string" ] },
+                 then: "$member"
+               },
+               { case:
+                 { $eq: [ { $type: "$member" }, "bool" ] },
+                 then: {
+                   $cond: { if: "$member",
+                     then: "Gold",
+                     else: "Guest" } } 
+                 },
+               { case:
+                 { $eq: [ { $type: "$member" }, "array" ] },
+                 then: { $last: "$member" }
+               }
+             ],
+             default: "Guest"}
+   }}}]
```

Comment:
I think the formatting makes this harder to follow what is going on. For example, the { after case is broken onto the same level as case, but after then (which is on the same level as case), the line is broken after the {, and then the $cond is indented. The } after default is collapsed to that same line, but not the one after that last `then`.

I'm not sure what the convention should be, but maybe something like:
1. To start, expand everything using a typical json formatter.
2. If a "level" is collapsed, it should all be put one one line: `case: { $eq: [ { $type: "$member" }, "string" ] },`.
3. Where it cannot be put on one line, but collapsing is helpful, keep all ({[s strictly matching some line they were introduced on. So, `[ { $project: {` is an acceptable collapse, but the trailing `} } ]` must be on a similarly-indented line at the bottom, with nothing else on that line.
4. When choosing indentation, attempt to mimic what was done in the Java, for the sake of easier comparison.